### PR TITLE
Allow 4 concurrent delete operations

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description

Within the trashbin up to 4 permanent delete operations are performed in parallel at a time.
This approach needs to be adapted to any file operation ....

## Related Issue
#1502 

## How Has This Been Tested?
- delete a lot of files
- go to trash bin
- mark some of them (10+)
- open network tab
- click delete selected
- see 4 requests being executed in parallel
Screenshot from 2019-07-18 11-11-38

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4199/10

<p>





</p></details>

**Comment by DeepDiver1975** [comment]
requires :
- [ ] https://github.com/owncloud/owncloud-sdk/pull/289
- [ ] https://github.com/owncloud/core/pull/35879

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4245/9

<p>










</p></details>

File: apps/files/src/components/Trashbin.vue [hunk 1]
```diff
@@ -123,26 +131,45 @@ export default {
...
             })
-          })
-          .catch(error => {
-            const translated = this.$gettext('Deletion of %{file} failed')
-            this.showMessage({
-              title: this.$gettextInterpolate(translated, { file: file.name }, true),
-              desc: error.message,
-              status: 'danger'
+            .catch(error => {
+              if (error.statusCode && error.statusCode === 423) {
```
Comment (LukasHirt) [inline apps/files/src/components/Trashbin.vue]: Does here need to be the first condition? Don't we already know if the statusCode exists if the second condition is true?

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4251/9

<p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4251/10

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4309/10

<p>





</p></details>

**Comment by DeepDiver1975** [comment]
Needs Server pr ..... 🙈

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4418/10

<p>







</p></details>

**Comment by LukasHirt** [comment]
> Needs Server pr ..... 🙈

Which one does it need? Or is it already merged to stable10? Still having issues
- Files get deleted in the end but still getting locked errors.

This is how looks my network after deleting 5 files:
image

- confirmation dialog is not being closed.

**Comment by DeepDiver1975** [comment]
> Which one does it need? Or is it already merged to stable10? Still having issues

already merged by now

**Comment by DeepDiver1975** [comment]
>     * Files get deleted in the end but still getting locked errors.

yes - this is the trick - we retry the operation until is succeeds

**Comment by DeepDiver1975** [comment]
>     * confirmation dialog is not being closed.

not addressed in this PR - needs fixing in a separate PR

**Comment by LukasHirt** [comment]
> > ```
> > * Files get deleted in the end but still getting locked errors.
> > ```
> 
> yes - this is the trick - we retry the operation until is succeeds

Ah, I get it now. Got a little bit confused. Then it's fine :grin:

**Comment by DeepDiver1975** [comment]
I will have a look at the failing tests tomorrow

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4436/10

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4439/10

<p>





</p></details>

**Pull request merged**